plugins/VisitFrequency/API.php

```diff
@@ -36,48 +40,80 @@ class API extends \Piwik\Plugin\API
     public function get($idSite, $period, $date, $segment = false, $columns = false)
     {
         Piwik::checkUserHasViewAccess($idSite);
-        $segment = $this->appendReturningVisitorSegment($segment);
-
-        $this->unprefixColumns($columns);
-        $params = array(
-            'idSite'    => $idSite,
-            'period'    => $period,
-            'date'      => $date,
-            'segment'   => $segment,
-            'columns'   => implode(',', $columns),
-            'format'    => 'original',
-            'format_metrics' => 0
+
+        $visitTypes = array(
+            self::NEW_COLUMN_SUFFIX => self::NEW_VISITOR_SEGMENT,
+            self::RETURNING_COLUMN_SUFFIX => self::RETURNING_VISITOR_SEGMENT
         );
 
-        $table = Request::processRequest('VisitsSummary.get', $params);
-        $this->prefixColumns($table, $period);
-        return $table;
+        $columns = Piwik::getArrayFromApiParameter($columns);
+
+        /** @var \Piwik\DataTable\DataTableInterface $resultSet */
+        $resultSet = new DataTable();
```

Comment:
FYI, this needs to be `new Simple()` otherwise the XML output will have an extra `<row>` element.

Reply:
cheers @diosmosis